.github/workflows/updatereleaseversion.yml

```diff
@@ -0,0 +1,35 @@
+name: Update release version
+on:
+  pull_request:
+    branches: [ master ]
+    types:
+      - opened
+      - reopened
+
+env:
+  GITHUB_TOKEN: ${{ github.token }}
+
+jobs:
+  build:
+    runs-on: windows-latest
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      with:
+        repository: ${{ github.event.pull_request.head.repo.full_name }}
+        ref: ${{ github.event.pull_request.head.ref }}
+      
+    - name: Update manifest.json version number
+      uses: jossef/action-set-json-field@v1
+      with:
+        file: src/manifest.json
+        field: version
+        value: "0.55.5.${{ github.run_number }}"
```

Comment:
Won't we have to change the version number every version bump?

Reply:
This workflow currently updates the most minor version for every pull request and automatically bumps it. We can change it to update other ways as well, I don't know how much we want to increase it by default so I went with the smallest automatic change by default.

Reply:
We may also want to update the current_version.json file as well but I don't have that in my project so I don't know what all in there needs to be addressed since I am unsure why it has a separate mac os version.

Reply:
The first automatically built package is the following:
https://github.com/GOG-Nebula/galaxy-integration-uplay/releases/tag/early_release_0.55.5.1

Reply:
Oh I realise that the purpose of this is basically to auto create pre release releases for people to test. Whenever a new version comes out then we manually bump it and release it.

Reply:
We can also probably add an automated more major version increase for full releases if we wanted it but I would have to look into it more to determine the details.

Reply:
yeah for now we can just do it manually ig

Reply:
What would we want to do (possibly differently) for the full releases?
I would expect:
- removing the pre-release notation and marking it as release
- updating the release number
- tagging
- packing up a built version on the releases tab

That last step seems to be different on other plugins since they have a tasks.py file setup for packaging the files while this one does not. I included all dependencies (for windows) into the zip file while some other ones explicitly exclude them. We could update the build to be like these packages with the tasks.py file and include the generated file in the releases instead. 

Updating the release number with this workflow is only done when generating a pull request and with minimal increases based on the github run of the workflow, since I only touch any plugins' master branches through pull requests, but that can be changed to other methodologies if we wish. An example change could be to set all pushes into master as final releases and not mark the tag and package as pre-release.

I don't know what other follow-on processes we would want.

Reply:
> That last step seems to be different on other plugins since they have a tasks.py file setup for packaging the files while this one does not. I included all dependencies (for windows) into the zip file while some other ones explicitly exclude them. We could update the build to be like these packages with the tasks.py file and include the generated file in the releases instead.

Yeah we could use `tasks.py` to create a build script for the repo using `buildtools` and then the Github action could use that to create win and mac packages.

> Updating the release number with this workflow is only done when generating a pull request and with minimal increases based on the github run of the workflow, since I only touch any plugins' master branches through pull requests, but that can be changed to other methodologies if we wish. An example change could be to set all pushes into master as final releases and not mark the tag and package as pre-release.

Maybe we should have a dev branch. I'm not quite sure about automating a version bump (not sure how to implement it and may cause issues). We could implement a system where we have a `dev` branch where we merge our PRs to. Each PR would include a pre-release version bump (if we follow https://semver.org/ it would look something like `0.55.6-alpha.2` or `0.55.6-beta` or `0.55.6-rc.3`. See https://github.com/semver/semver/issues/114#issuecomment-19366348 for the meanings). After we think the rc release is ready we can create a PR to merge `dev` into `master` bumping the version in `dev` from `0.55.6-rc.3` to `0.55.6` and `master` from `0.55.5` to `0.55.6`. This would become tedious if we were working together on a feature or fix and we could setup org rules that allow us to create branches on the repo which we have write access to (e.g. `feature/achievements`) and then we would create a PR to merge that into `dev`. However, this isn't automated and is possibly too tedious or complicated practically but just putting the idea out there.